### DISCUSSION TITLE
Temperature problem

### DISCUSSION THREAD

**Discussion by Brian9924** [firemodels/fds#11666] [General]
Hello, everyone! I would like to ask an issue that in a full size shaft fire, why the temperature will drop below 0℃ with 0.2m size of mesh. 

![1681098892356](https://user-images.githubusercontent.com/120773534/230822621-2b0d4f19-556e-4998-bb05-7976fc3a8df6.jpg)
![1](https://user-images.githubusercontent.com/120773534/230822312-a1f073b3-de9d-48a1-8ef1-dc3f850eee1e.png)
![2](https://user-images.githubusercontent.com/120773534/230822316-a8f07321-3608-4e95-a732-645aca586378.png)
![3](https://user-images.githubusercontent.com/120773534/230822323-6752b395-3a2f-436e-abe8-c0ee2acab648.png)
![0 2m](https://user-images.githubusercontent.com/120773534/230822378-07940fba-5ae4-4fef-95d5-037395ae92de.jpg)
![0 25m](https://user-images.githubusercontent.com/120773534/230822383-eb94d23f-bb9f-4c17-82c9-594a2599f7db.jpg)

**Comment by rmcdermo** [comment]
We need an input file.

**Comment by Brian9924** [comment]
Thanks for your comment, these are the input files:
1.Mesh 0.2m:
[1_6_0_8_0.2m.txt](https://github.com/firemodels/fds/files/11190651/1_6_0_8_0.2m.txt)
2.Mesh 0.25m:
[1_6x0_8_0.25m.txt](https://github.com/firemodels/fds/files/11190666/1_6x0_8_0.25m.txt)

**Comment by rmcdermo** [comment]
Just getting time to look at this.  First, you really have too many cells for one or two meshes.  Even in the 2 mesh case, I have been running for 20 minutes and have not produced a single time step yet.  This case needs parallel processing.

But the issue you are seeing is most likely related to the use of 2 meshes in the 0.2 m case.  Add this line to your input file and try the case again.

&PRES VELOCITY_TOLERANCE=1.E-6, MAX_PRESSURE_ITERATIONS=100/

Let me know if this solves the problem.

**Comment by Brian9924** [comment]
Thank you for your suggestion, I will give it a go.

**Comment by Brian9924** [comment]
Dear Mr. McDermott @rmcdermo , I have added the code which you have mentioned  to input file. And I divided eight meshes again. But it seems that only Mesh 01(MPI 0) outputs the results. The results of other meshes are invariant. How can I solve this problem？ THANK YOU！
![image](https://user-images.githubusercontent.com/120773534/231396739-9b642a27-2d0f-45d5-bfb9-1b96b6e670b3.png)
![image](https://user-images.githubusercontent.com/120773534/231397453-518ca792-55e7-45cf-992d-79d94b6f985c.png)
![image](https://user-images.githubusercontent.com/120773534/231397969-12adea61-64bb-42a1-9b35-7b0cd2ab1cf1.png)
![image](https://user-images.githubusercontent.com/120773534/231398084-b52a7255-2bdc-4d2e-a7fe-8d70eced7229.png)
![image](https://user-images.githubusercontent.com/120773534/231398539-243ea561-b496-4114-a676-2a75a3620902.png)
input file：
[0.2.txt](https://github.com/firemodels/fds/files/11208959/0.2.txt)

**Comment by rmcdermo** [comment]
I am not getting negative temperatures after adding the PRES line.

I'm running your 8 mesh case now.  Did you look at the SLCF and compare with the DEVC positions?  You are only 18 seconds into the simulation; isn't it possible that the fire has not reached those DEVC positions yet?

**Comment by Brian9924** [comment]
Thank you for your patience.
Maybe I use MPI and OpenMP incorrectly? It is seen that there is not any information interaction between Mesh01 and other meshes.
![1681304010572](https://user-images.githubusercontent.com/120773534/231463388-1d8b429c-7ff0-4063-803a-cdd34bf06a9c.jpg)
![image](https://user-images.githubusercontent.com/120773534/231461744-3b7c59cc-7154-43b2-af91-b58cca795757.png)
[0_2_hrr.csv](https://github.com/firemodels/fds/files/11211660/0_2_hrr.csv)
[0_2_steps.csv](https://github.com/firemodels/fds/files/11211661/0_2_steps.csv)
[0_2_devc.csv](https://github.com/firemodels/fds/files/11211662/0_2_devc.csv)

**Comment by rmcdermo** [comment]
Sorry, I transferred this case to Discussions and it missed the comments I had made, so here I'll restate them.

This case appears to be running as designed and so I do not see this as an Issue.  The DEVC values are reporting the same as the SLCF values.

You have very coarse grid resolution.  And if your shaft is sealed, this can choke the flow, keeping upper temperatures artificially low.  I recommend you increase resolution and that you ensure you have the proper ventilation paths set in your case.

You should also use only MPI for this case and map all the physical cores on your computer to a mesh.  Look at the user guide for how to set OpenMP threads to 1 or just use the non-OpenMP executable.

<img width="1603" alt="Screenshot 2023-04-12 at 9 34 57 AM" src="https://user-images.githubusercontent.com/4418497/231477711-d91311b5-5592-4c08-a03a-3a208f5ee33e.png">

**Reply by Brian9924** [reply]
Thank you for your guidance. It is really helpful to me!

**Comment by rmcdermo** [comment]
Well, maybe I spoke too soon.  There is something odd going on at the mesh boundary.  There is clearly a blockage there that I suspect you don't want.  I'm sending this back over the @mcgratta .

The case is located here: `/home/rmcdermo/Issues/Issue_11655`  The input file is `test_8mesh.fds`.

<img width="1215" alt="Screenshot 2023-04-12 at 4 24 47 PM" src="https://user-images.githubusercontent.com/4418497/231576164-c6e85e97-879e-4d9d-8bd7-e050eebb36b6.png">

**Comment by Brian9924** [comment]
There is a floor exactly at the edge of the mesh01，and the hole didn't go through this floor. So I readjusted the mesh01. Now, the results are normal.
![3022d6f84636eaa7f58f2a701765807](https://user-images.githubusercontent.com/120773534/231743417-534ef4f9-8940-4d2f-9337-631128cc3f9b.png)
![64e65d70b2d511b0873e1387851ab5b](https://user-images.githubusercontent.com/120773534/231743437-dd2a4397-bead-4634-8377-23e170105b82.png)

**Comment by rmcdermo** [comment]
OK, thanks for reporting.
